types/react-modal/index.d.ts

```diff
@@ -50,11 +50,11 @@ declare namespace ReactModal {
         /* String className to be applied to the portal. Defaults to "ReactModalPortal". */
         portalClassName?: string;
 
-        /* String className to be applied to the document.body. */
-        bodyOpenClassName?: string;
+        /* String className to be applied to the document.body (must be a constant string). When set to null it doesn't add any class to document.body. */
+        bodyOpenClassName?: string | null;
```

Comment:
Would this be clearer as:

```suggestion
        bodyOpenClassName: string | null | undefined;
```

Can this (and the other changed properties) actually be both `null` and `undefined`?

Reply:
I think the `bodyOpenClassName` is optional, so optional parameter (`bodyOpenClassName?`) is semantically correct.

The end result is the same, `?` adds the `undefined` to the type, but at least for me, it is easier to read the type if I can quickly glance what is mandatory and what is optional instead of searching `undefined` from the list of types separated by pipe operators.

Reply:
All of them are indeed optional but nullable. In which case I tend to prefer the optional notation as well.

Reply:
Removing the `?` is definitely not what you want. It would force users to supply the prop on JSX elements 100% of the time, even if to set to `undefined`: https://www.typescriptlang.org/play/?jsx=2#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4BYAKCpgE8wk4AFHMAZzgF44BvONFADsAQkgCqggCZJMwQUkkAuOGxhQ5AczgAfOAFcpMuQrgBfCpWloANiiJxMBjMAiC4AYVyR5gmAAowVjZlFgh2AEplACkAZQANADoAUWskECRfCzRXVX4uOAAeT3BXDPgAegA+LJz4SXyir1LffiFRCWlZeUlOHgNO40lTOCrSIA